l2-contracts/contracts/ConsensusAuthority.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import "./ValidatorRegistry.sol";
+import "./AttesterRegistry.sol";
+
+contract ConsensusAuthority {
+    // Owner of the contract (i.e., the governance contract).
+    address public owner;
+    // The validator registry instance.
+    ValidatorRegistry public validatorRegistry;
+    // The attester registry instance.
+    AttesterRegistry public attesterRegistry;
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Unauthorized: onlyOwner");
+        _;
+    }
+
+    modifier onlyOwnerOrNodeOwner(address nodeOwner) {
+        require(msg.sender == owner || msg.sender == nodeOwner, "Unauthorized: onlyOwnerOrNodeOwner");
+        _;
+    }
+
+    constructor(address _owner) {
```

Comment:
not sure if we plan on updating this in the future but it may be worth making these contracts upgradeable

Reply:
The current plan is to redirect the contract pointer to a different address once an upgrade will be needed.

Reply:
that makes sense, for the 2 registries though if they're not upgradeable you'd have to migrate the mappings and arrays. unless the plan was to use the force deployer to deploy to the same address

Reply:
The plan is to not require state migration / persistence while upgrading. Once that will be needed, there will be an upgradable version for this contract.